manim/scene/graph_scene.py

```diff
@@ -132,17 +153,28 @@ def setup_axes(self, animate=False):
         if self.y_axis_label:
             y_label = TextMobject(self.y_axis_label)
             y_label.next_to(
-                y_axis.get_corner(UP + RIGHT), UP + RIGHT,
+                y_axis.get_corner(self.y_label_position), self.y_label_position,
                 buff=SMALL_BUFF
             )
             y_label.shift_onto_screen()
             y_axis.add(y_label)
             self.y_axis_label_mob = y_label
 
-        if animate:
-            self.play(Write(VGroup(x_axis, y_axis)))
-        else:
-            self.add(x_axis, y_axis)
+        if self.x_axis_visibility and self.y_axis_visibility:
+            if animate:
+                self.play(Write(VGroup(x_axis, y_axis)))
+            else:
+                self.add(x_axis, y_axis)
+        elif self.x_axis_visibility:
+            if animate:
+                self.play(Write(x_axis))
+            else:
+                self.add(x_axis)
+        elif self.y_axis_visibility:
+            if animate:
+                self.play(Write(y_axis))
+            else:
+                self.add(y_axis)
```

Comment:
make this more DRY
```suggestion
        axes = []
        if self.x_axis_visibility:
            axes.append(x_axis)
        if self.y_axis_visibility:
            axes.append(y_axis)

        if animate:
            self.play(Write(VGroup(*axes)))
        else:
            self.add(*axes)
```

Reply:
Thanks for that. I actually wasn't quite happy with the structure, but couldn't manage to come up with your method